assets/js/components/Portal.js

```diff
@@ -20,18 +20,28 @@
  * External dependencies
  */
 import { useEffectOnce } from 'react-use';
+import PropTypes from 'prop-types';
 
 /**
  * WordPress dependencies
  */
 import { createPortal, useState } from '@wordpress/element';
 
-function Modal( { children } ) {
+function Portal( { children, slug, appendPluginRoot } ) {
+	const elDiv = document.createElement( 'div' );
+	if ( slug ) {
+		elDiv.classList.add( `googlesitekit-portal-${ slug }` );
+	}
+
+	if ( ! appendPluginRoot ) {
+		elDiv.classList.add( 'googlesitekit-plugin' );
+	}
+
 	// Using state as we need `el` to not change when the component re-renders
-	const [ el ] = useState( document.createElement( 'div' ) );
+	const [ el ] = useState( elDiv );
```

Comment:
I think we can move class list manipulations to the `useEffectOnce` hook:

```js
const [ el ] = useState( document.createElement( 'div' ) );

useEffectOnce( () => {
    if ( slug ) {
        el.classList.add( `googlesitekit-portal-${ slug }` );
    }

    if ( ! appendPluginRoot ) {
        el.classList.add( 'googlesitekit-plugin' );
    }

    const root = appendPluginRoot
        ? document.querySelector( '.googlesitekit-plugin' ) || document.body
        : document.body;

    root.appendChild( el );

    return () => root.removeChild( el );
} );
```

Reply:
👍